bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerMetadata.java

```diff
@@ -84,12 +85,14 @@
 
     private boolean hasPassword = false;
     private LedgerMetadataFormat.DigestType digestType;
+    private LedgerMetadataFormat.LedgerType ledgerType;
     private byte[] password;
 
     private Map<String, byte[]> customMetadata = Maps.newHashMap();
 
     public LedgerMetadata(int ensembleSize, int writeQuorumSize, int ackQuorumSize,
```

Comment:
Can you create an issue for using Builder pattern to construct ledger metadata? we should use builder pattern here.

Reply:
may it could be done as part of making the metadata immutable. I had a patch for it before I left yahoo, but it was never merged. I could dig it out, or maybe just redo. It was only a day or two of work if i remember correctly.

Reply:
@ivankelly  thank you I will let you drive this additional change